api/src/registry.rs

```diff
@@ -215,35 +223,22 @@ impl ImageConfig {
 }
 
 #[derive(Debug, Clone, Fail)]
-pub enum ImagesServiceError {
-    #[fail(display = "Unexpected docker registry error: {}", internal_message)]
-    UnexpectedError { internal_message: String },
-    #[fail(display = "Unexpected docker image blob format: {}", internal_message)]
-    InvalidImageBlob { internal_message: String },
-}
-
-impl From<IOError> for ImagesServiceError {
-    fn from(e: IOError) -> Self {
-        ImagesServiceError::UnexpectedError {
-            internal_message: format!("{}", e),
-        }
-    }
-}
-
-impl From<DKRegistryError> for ImagesServiceError {
-    fn from(e: DKRegistryError) -> Self {
-        ImagesServiceError::UnexpectedError {
-            internal_message: format!("{}", e),
-        }
-    }
-}
-
-impl From<serde_json::Error> for ImagesServiceError {
-    fn from(e: serde_json::Error) -> Self {
-        ImagesServiceError::InvalidImageBlob {
-            internal_message: format!("{}", e),
-        }
-    }
+pub enum RegistryError {
```

Comment:
Also on a general note when you launch the application on cargo run -- --runtime-type Kubernetes and start a new app with any payload like below.
`[
  {
    "serviceName": "postgres",
    "image": "postgres:latest",
    "env": {
      "MYSQL_USER": "admin"
    },
    "files": {
      "/etc/mysql/my.cnf": "[client-server]\n# Uncomment these if you want to use a nonstandard connection to MariaDB\n#socket=/tmp/mysql.sock\n#port=3306\n"
    }
  }
]`

Kube returns a `[2024-01-10T11:06:31Z WARN  kube_client::client] Unsuccessful data error parse: 404 page not found`
Which is translated by Routes as an Error but the app is started anyways.
`[2024-01-10T11:06:31Z ERROR prevant::apps::routes] Internal server error: Cannot interact with infrastructure: Unexpected Kubernetes interaction error: ApiError: "404 page not found\n": Failed to parse error data (ErrorResponse { status: "404 Not Found", message: "\"404 page not found\\n\"", reason: "Failed to parse error data", code: 404 })`

This could be slightly misleading.

Reply:
I couldn't reproduce this issue.

Reply:
However, I included a working example in the payload. Maybe, you want to check that out as well.

Reply:
Fixed.